sql/core/src/main/scala/org/apache/spark/sql/execution/basicOperators.scala

```diff
@@ -200,6 +204,10 @@ case class TakeOrderedAndProject(
     projectOutput.getOrElse(child.output)
   }
 
+  override def outputsUnsafeRows: Boolean = true
```

Comment:
What you have done when `projectList` is `None` is exactly same with `ConvertToUnsafe` right? How about we change this to `if (projectList.isDefined) true else child.outputsUnsafeRows`, then our framework can insert `ConvertToUnsafe` if it's necessary.